File: Prototype_Apache4/src/java/aa/Bid.java
package aa;

import java.util.Date;

// represents a bid (in a buy order)
public class Bid {

  private String stock;
  private int price; // bid price
  private String userId; // user who made this buy order
  private Date date;

  // constructor
  public Bid(String stock, int price, String userId) {
    this.stock = stock;
    this.price = price;
    this.userId = userId;
    this.date = new Date();
  }
  
  public Bid(String stock, int price, String userId,Date date) {
    this.stock = stock;
    this.price = price;
    this.userId = userId;
    this.date = date;
  }
  
  // getters
  public String getStock() {
    return stock;
  }

  public int getPrice() {
    return price;
  }

  public String getUserId() {
    return userId;
  }

  public Date getDate() {
    return date;
  }

  // toString
  public String toString() {
    return "stock: " + stock + ", price: " + price + ", userId: " + userId + ", date: " + date;
  }
}

File: AA_Domain/routes/modules/bid.js
function Bid (idvalue, stock, price, userId, date) {
	this.idvalue = idvalue;
	this.stock = stock;
	this.price = price;	
	this.userId = userId;
	this.date = date;
}

Bid.prototype.getStock = function() {
	return this.stock;
}

Bid.prototype.getPrice = function() {
	return this.price;
}

Bid.prototype.getUserId = function() {
	return this.userId;
}

Bid.prototype.getDate = function() {
	return this.date;
	//return this.date.getMonth() + 1 + "-" + this.date.getDate() + "-" + this.date.getFullYear();
}

Bid.prototype.getId = function() {
	return this.idvalue;
}

Bid.prototype.setId = function(idvalue) {
	this.idvalue = idvalue;
}

Bid.prototype.getDateString = function() {
	var dateString = [[AddZero(this.date.getDate()), AddZero(this.date.getMonth() + 1), this.date.getFullYear()].join("/"), [AddZero(this.date.getHours()), AddZero(this.date.getMinutes())].join(":"), this.date.getHours() >= 12 ? "PM" : "AM"].join(" ");
	return dateString;
}

Bid.prototype.toString = function() {
	var dateString = [[AddZero(this.date.getDate()), AddZero(this.date.getMonth() + 1), this.date.getFullYear()].join("/"), [AddZero(this.date.getHours()), AddZero(this.date.getMinutes())].join(":"), this.date.getHours() >= 12 ? "PM" : "AM"].join(" ");
	return "stock: " + this.stock + ", price: " + this.price + ", userId: " + this.userId + ", date: " + dateString;
}

//Pad given value to the left with "0"
function AddZero(num) {
    return (num >= 0 && num < 10) ? "0" + num : num + "";
}

module.exports.Bid = Bid;